src/Sarif.Driver/Sdk/ExportRulesMetadataCommandBase.cs

```diff
@@ -117,37 +117,17 @@ private void OutputSarifRulesMetada(string outputFilePath, ImmutableArray<IRule>
             run.Results = new List<Result>();
 
             log.Runs.Add(run);
-            run.Resources = new Resources
-            {
-                Rules = new List<Rule>()
-            };
-
-            SortedDictionary<int, Rule> sortedRules = new SortedDictionary<int, Rule>();
-
-            foreach (IRule rule in skimmers)
-            {
-                var newRule = new Rule();
```

Comment:
>newRule [](start = 20, length = 7)

Weird. Do you remember what this was for?

Reply:
Yes, note that the incoming param was an IRule. We had to convert to a concrete rule instance for some reason (to get access to downstream API). This horribleness drove the extirpation of IRule, now completely departed.

---
In reply to: [255311248](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255311248) [](ancestors = 255311248)